src/main/java/org/primefaces/extensions/renderer/CommandButtonSingleClickRenderer.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.renderer;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import org.primefaces.component.commandbutton.CommandButton;
+import org.primefaces.component.commandbutton.CommandButtonRenderer;
+
+/**
+ * {@link CommandButton} renderer disabling the button while action is processed.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ * @since 8.0
+ */
+public class CommandButtonSingleClickRenderer extends CommandButtonRenderer {
+
+    protected static final String[] CLASS_CONFIRM = {"ui-confirmdialog-yes", "ui-confirmdialog-no"};
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
```

Comment:
Better to override encodeMarkup instead no?

Reply:
Why? I don't think it'll make much of a difference.

Reply:
It doesn't indeed, but `encodeMarkup` is meant for HTML stuff

Reply:
We are modifying attributes and don't really render anything.

Reply:
exactly, HTML attributes

Reply:
Alright, done. Also saved one line 😄